src/Compilers/Core/Portable/Emit/EditAndContinue/DeletedMethodDefinition.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable disable
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Reflection;
+using Microsoft.Cci;
+using Microsoft.CodeAnalysis.Symbols;
+
+namespace Microsoft.CodeAnalysis.Emit.EditAndContinue
+{
+    internal sealed partial class DeletedMethodDefinition : IMethodDefinition
+    {
+        private IMethodSymbolInternal _methodDef;
+        private IMethodDefinition _oldMethod;
+
+        public DeletedMethodDefinition(IMethodDefinition oldMethod, IMethodSymbolInternal methodDef)
+        {
+            _oldMethod = oldMethod;
+            _methodDef = methodDef;
+        }
+
+        public IEnumerable<IGenericMethodParameter> GenericParameters => _oldMethod.GenericParameters;
+
+        public bool HasDeclarativeSecurity => _oldMethod.HasDeclarativeSecurity;
+
+        public bool IsAbstract => _oldMethod.IsAbstract;
+
+        public bool IsAccessCheckedOnOverride => _oldMethod.IsAccessCheckedOnOverride;
+
+        public bool IsConstructor => _oldMethod.IsConstructor;
+
+        public bool IsExternal => _oldMethod.IsExternal;
+
+        public bool IsHiddenBySignature => _oldMethod.IsHiddenBySignature;
+
+        public bool IsNewSlot => _oldMethod.IsNewSlot;
+
+        public bool IsPlatformInvoke => _oldMethod.IsPlatformInvoke;
+
+        public bool IsRuntimeSpecial => _oldMethod.IsRuntimeSpecial;
+
+        public bool IsSealed => _oldMethod.IsSealed;
+
+        public bool IsSpecialName => _oldMethod.IsSpecialName;
+
+        public bool IsStatic => _oldMethod.IsStatic;
+
+        public bool IsVirtual => _oldMethod.IsVirtual;
+
+        public ImmutableArray<IParameterDefinition> Parameters => _oldMethod.Parameters;
```

Comment:
>_oldMethod.Parameters [](http://example.com/codeflow?start=66&length=21)

Are these accessors that return references to the old compilation being called? If so, it still feels like there's a chance the metadata writer could get incorrect info. If not,  perhaps these accessors should `throw ExceptionUtilities.Unreachable` instead.